### DISCUSSION TITLE
Add a simple way for a module to require a minimum version of PHP

### DISCUSSION THREAD

**Discussion by Huluti** [PrestaShop/PrestaShop#33417] [Ideas]
#### Is your feature request related to a problem?

As a module developer, we need to prevent installs and upgrade to a module that will requires PHP 7.1 at least for example.

#### Describe the solution you'd like

It would be good to have a simple way to specify this prerequisite and don't allow installs or upgrades.

#### Alternatives you've considered

I've managed to detect versions of PHP used with:
`if (version_compare(PHP_VERSION, '7.1.0', '<')) {`

But I can't managed to disallow an install or an upgrade.

#### Additional context

In WordPress for example, we can add a 'Requires PHP' header to the readme.txt file of the plugin.

**Comment by hibatallahAouadni** [comment]
Hello @Huluti 

Thank you for your suggestion. The Developer Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

PS: ping @matks I think you may like to discuss it :relaxed: 

Thank you

**Comment by matks** [comment]
Hello @Huluti 

We provide a map of "what versions of PHP are compatible with PrestaShop versions" on https://devdocs.prestashop.com/1.7/basics/installation/system-requirements/

So theoretically, if I acknowledge that module A is compatible with PrestaShop 1.7.4, it _has_ to be compatible with 5.6 to 7.1

If I acknowledge that module A is compatible with PrestaShop 1.7.7, it _has_ to be compatible with 7.1 to 7.3

**Comment by Huluti** [comment]
Hi @matks! Ah yes indeed, so the easiest way would be to limit the compatibility of the plugin with this line: ps_versions_compliancy and and also to limit the releases on the addons/prestashop site in our seller account? It seems to me to be viable and this issue isn't one so. Thank's for your time!

**Comment by Huluti** [comment]
But for example, our plugin is compatible 1.6 and 1.7, but the plugin could be installed on multiple PHP versions while we need to be sure that the user have PHP 7.1 at least  (because our SDK is PHP 7.1+). So my previous comment is not valid in our case...

**Comment by matks** [comment]
> But for example, our plugin is compatible 1.6 and 1.7, but the plugin could be installed on multiple PHP versions while we need to be sure that the user have PHP 7.1 at least (because our SDK is PHP 7.1+). So my previous comment is not valid in our case...

I see.

In my opinion it makes sense. I would like a 2nd review, but I would say it's a relevant module feature request.

ping @PrestaShop/prestashop-maintainers

**Comment by Huluti** [comment]
Thanks for your return. I think that in addition it's could be good to encourage people keeping their environment up to date.

PS: Could be useful also if the requirement was added in the add-on developer account.

**Comment by kpodemski** [comment]
I think that the entire process of uploading/installing the module could be a bit different.

Right now, when we upload the new module, it is replacing the old one. Maybe we should update it to the temporary location, read some `.yml`, use `ReflectionClass` or something, and validate if every requirement has been met? then we can move it to the destinated location
